keyboards/plaid/keymaps/thehalfdeafchef/keymap.c

```diff
@@ -0,0 +1,237 @@
+ * | NLCK |  1   |  2   |  3   |  -   |     |     |     |  !   |  #   |  $   |  +   |
+ * |------+------+------+------+------+-----+-----+-----+------+------+------+------+
+ * |      |  4   |  5   |  6   |  +   |     |     |     |  %   |  !   |  !   |  =   |
+ * |------+------+------+------+------+-----+-----+-----+------+------+------+------+
+ * |      |  7   |  8   |  9   |  *   |     |     |     |  _   |  -   |  @   |  ~   |
+ * |------+------+------+------+------+-----+-----+-----+------+------+------+------+
+ * |      |  0   |  .   | Ent  |  =   |           |     |      |      |      |  `   |
+ * `--------------------------------------------------------------------------------'
+ */
+
+[_RAISE] =
+LAYOUT_planck_mit(
+        KC_NLCK, KC_P1, KC_P2, KC_P3, KC_PMNS, KC_NO, KC_NO, KC_NO, KC_EXLM, KC_HASH, KC_DLR, KC_PLUS,
+        KC_PSLS, KC_P4, KC_P5, KC_P6, KC_PPLS, KC_NO, KC_NO, KC_NO, KC_PERC, KC_ASTR, KC_AMPR, KC_EQL,
+        KC_PAST, KC_P7, KC_P8, KC_P9, KC_PCMM, KC_NO, KC_NO, KC_NO, KC_UNDS, KC_MINS, KC_AT, KC_TILDE,
+        KC_NO, KC_P0, KC_PDOT, KC_PENT, KC_PEQL, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_GRV
+),
+
+/* Function (Lower + Raise)
+ * ,-----------------------------------------------------------------------------------------------------------------------.
+ * | RESET| LCA_T(F1)  | LCA_T(F2) | LCA_T(F3) | LCA_T(F4) | LCA_T(F5) | LCA_T(F6) | LCA_T(F7) |      |      |      |      |
+ * |------+------------+-----------+-----------+-----------+-----------+-----------+-----------+------+------+------+------+
+ * |      |            |           |           |           | QWERTY    | DVORAK    | COLEMAK   |      |      |      |      |
+ * |------+------------+-----------+-----------+-----------+-----------+-----------+-----------+------+------+------+------+
+ * |      |            |           |           |           |           |           |           |      |      |      |      |
+ * |------+------------+-----------+-----------+-----------+-----------+-----------+-----------+------+------+------+------+
+ * |      |            |           |           |           |          LED          |           |      |      |      |      |
+ * `-----------------------------------------------------------------------------------------------------------------------'
+ */
+
+[_FUNCTION] =
+LAYOUT_planck_mit(
+        RESET, LCA_T(KC_F1), LCA_T(KC_F2), LCA_T(KC_F3), LCA_T(KC_F4), LCA_T(KC_F5), LCA_T(KC_F6), LCA_T(
+        KC_F7), KC_NO, KC_NO, KC_NO, KC_NO,
+        KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, QWERTY, DVORAK, COLEMAK, KC_NO, KC_NO, KC_NO, KC_NO,
+        KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO,
+        KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, LED, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO
+)
+};
+
+// constants to toggle LED behavior
+#define LED_ON 1
+#define LED_OFF 0
+
+
+//Setup config struct for LED
+typedef union {
+    uint32_t raw;
+    struct {
+        uint8_t red_mode :8;
```

Comment:
You're only using booleans for this?   If so, you don't need the whole 8 bits here. 

```suggestion
        bool red_mode :1;
```